src/stratis/run.rs

```diff
@@ -83,10 +82,7 @@ async fn dm_event_thread(engine: Option<Arc<Mutex<dyn Engine>>>) -> StratisResul
                 dm_fd.next().await;
             }
         }
-        None => {
-            pending!();
```

Comment:
Unless I'm misunderstand the documentation of `select!()`, this will cause stratisd to always exit when the sim engine is in use. If one future resolves, `select!()` cancels all of the other futures. The only way to avoid this is to return a future that never resolves.

Reply:
I'll reread and take a look...the interesting thing is that the stratis-cli tests actually succeed, except for two that look like incoherence errors.